scripts/west_commands/build.py

```diff
@@ -253,13 +253,13 @@ def _setup_source_dir(self):
         if self.args.source_dir:
             source_dir = self.args.source_dir
         elif self.cmake_cache:
-            source_dir = self.cmake_cache.get('APPLICATION_SOURCE_DIR')
+            source_dir = self.cmake_cache.get('CMAKE_HOME_DIRECTORY')
```

Comment:
Should we use `CMAKE_SOURCE_DIR` instead?
And is this still valid when compiling an out-of-tree app?

Reply:
We should, but unfortunately it is not in the cache, so we must use the internal variable instead.